src/netrc.c

```diff
@@ -41,75 +41,95 @@
 #include <unistd.h>
 #include <sys/stat.h>
 #include <sys/mman.h>
+#include "axel.h"
 #include "netrc.h"
 
-#define REL_NETRC	"/.netrc"
-#define IS_DELIM(x)	(*x == ' ' || *x == '\t' || *x == '\n')
-
-static int fd;
-static size_t sz;
 static char *p;
 static char *s_addr;
 static char *e_addr;
 
-static size_t file_size(void)
+static size_t file_size(int fd)
 {
 	struct stat st;
 	fstat(fd, &st);
 	return st.st_size;
 }
 
-static int netrc_open(const char * const filename)
+static int netrc_mmap(const char *filename)
 {
-	int ret;
-	char *path;
-	char *netrc;
-	char *home;
-
-	ret = 0;
-	if (filename && filename[0] != '\0') {
-		path = strdup(filename);
-	} else {
-		if ((netrc = getenv("NETRC"))) {
-			path = strdup(netrc);
-		} else if ((home = getenv("HOME"))) {
-			size_t i = strlen(home) + strlen(REL_NETRC) + 1;
-			path = malloc(i);
-			snprintf(path, i, "%s%s", home, REL_NETRC);
-		} else {
-			return ret;
+	int fd;
+	size_t sz;
+	char *aux = NULL;
+	char *home = NULL;
+	char *path = NULL;
+	const char suffix[] = "/.netrc";
```

Comment:
I know, I know... But I thought this would be better than a `#define`. And if I used a `char *` instead of `char []`, that would imply usage of `strlen()` or hard-coded values.